spec/spec_helper.rb

```diff
@@ -46,14 +46,15 @@
 require './lib/capybara_timeout_patches'
 require 'pry'
 
+# BUG: Intermittent, seemingly non deterministic reversion of downloads directory
+#      from Capybara.save_path (./tmp/capybara) to project root. See:
+#      https://github.com/sanger/sequencescape/issues/3511
 Capybara.register_driver :headless_chrome do |app|
-  enable_chrome_headless_downloads(Capybara.drivers[:selenium_chrome_headless].call(app))
+  Capybara.drivers[:selenium_chrome_headless].call(app).tap { |driver| enable_chrome_headless_downloads(driver) }
 end
 
 def enable_chrome_headless_downloads(driver)
   driver.options[:options].add_preference(:download, default_directory: Capybara.save_path)
-  driver.browser.download_path = Capybara.save_path
-  driver
 end
```

Comment:
I lost a day to this problem, and didn't manage to get to the bottom of it. However I:
1) Created an issue so at least other people can track their findings
2) Removed some legacy stuff that was acting as a red-herring